File: scripts/create-default-languages.ts
import { db } from '../server/db';
import { languages } from '@shared/schema';

async function createDefaultLanguages() {
  try {
    console.log('Creating default languages...');

    const defaultLanguages = [
      { code: 'en', name: 'English', isActive: true },
      { code: 'pt', name: 'Portuguese', isActive: true },
      { code: 'es', name: 'Spanish', isActive: true },
      { code: 'fr', name: 'French', isActive: true },
      { code: 'de', name: 'German', isActive: true },
      { code: 'it', name: 'Italian', isActive: true },
      { code: 'ru', name: 'Russian', isActive: true },
      { code: 'ja', name: 'Japanese', isActive: true },
      { code: 'ko', name: 'Korean', isActive: true },
      { code: 'zh', name: 'Chinese', isActive: true }
    ];

    for (const lang of defaultLanguages) {
      try {
        await db.insert(languages).values(lang);
        console.log(`✅ Created language: ${lang.name} (${lang.code})`);
      } catch (error: any) {
        if (error.code === '23505') { // Unique constraint violation
          console.log(`⚠️  Language ${lang.name} (${lang.code}) already exists`);
        } else {
          console.error(`❌ Error creating language ${lang.name}:`, error.message);
        }
      }
    }

    console.log('\n🎉 Default languages created successfully!');
    console.log('\nAvailable languages:');
    const allLanguages = await db.select().from(languages);
    allLanguages.forEach(lang => {
      console.log(`- ${lang.name} (${lang.code})`);
    });

  } catch (error) {
    console.error('Error creating default languages:', error);
  } finally {
    process.exit(0);
  }
}

createDefaultLanguages();

File: backend/create_default_languages.py
#!/usr/bin/env python3

import os
import sys
import psycopg2

def create_default_languages():
    try:
        # Get database connection details from environment
        db_url = os.getenv('DATABASE_URL', 'postgresql://user:password@nativox-db:5432/nativox')
        conn = psycopg2.connect(db_url)
        cursor = conn.cursor()

        # Check if languages already exist
        cursor.execute("SELECT COUNT(*) FROM languages")
        count = cursor.fetchone()[0]
        
        if count > 0:
            print('✅ Languages already exist!')
            cursor.execute("SELECT id, name, code FROM languages")
            languages = cursor.fetchall()
            for lang in languages:
                print(f'  - {lang[1]} ({lang[2]}) - ID: {lang[0]}')
            return

        # Create default languages
        default_languages = [
            ('English', 'en'),
            ('Spanish', 'es'),
            ('Portuguese', 'pt'),
            ('French', 'fr'),
            ('German', 'de'),
            ('Italian', 'it'),
            ('Chinese', 'zh'),
            ('Japanese', 'ja'),
            ('Korean', 'ko'),
            ('Arabic', 'ar'),
            ('Russian', 'ru'),
            ('Hindi', 'hi'),
        ]

        for name, code in default_languages:
            cursor.execute("""
                INSERT INTO languages (name, code, is_active, created_at)
                VALUES (%s, %s, %s, NOW())
            """, (name, code, True))

        conn.commit()

        print('✅ Default languages created successfully!')
        for name, code in default_languages:
            print(f'  - {name} ({code})')

    except Exception as e:
        print(f'❌ Error creating default languages: {e}')
        if 'conn' in locals() and conn:
            conn.rollback()
    finally:
        if 'cursor' in locals() and cursor:
            cursor.close()
        if 'conn' in locals() and conn:
            conn.close()

if __name__ == "__main__":
    create_default_languages()
